src/Og.php

```diff
@@ -267,26 +252,28 @@ public static function membershipDefault() {
     return ['type' => 'og_membership_type_default'];
   }
 
+
   /**
-   * Get all the modules fields that can be assigned to fieldable entities.
+   * Get an OG field base definition.
    *
-   * @param $field_name
-   *   The field name that was registered for the definition.
+   * @param string $plugin_id
+   *   The plugin ID, which is also the default field name.
    *
+   * @throws \Exception
    * @return OgFieldBase|bool
-   *   An array with the field and instance definitions, or FALSE if not.
-   *
-   * todo: pass the entity type and entity bundle to plugin definition.
+   *   An array with the field storage config and field config definitions, or
+   *   FALSE if none found.
    */
-  protected static function fieldInfo($field_name = NULL) {
-    $config = \Drupal::service('plugin.manager.og.fields');
-    $fields_config = $config->getDefinitions();
+  protected static function getFieldBaseDefinition($plugin_id) {
+    /** @var OgFieldsPluginManager $plugin_manager */
+    $plugin_manager = \Drupal::service('plugin.manager.og.fields');
```

Comment:
Yes.  We will definitely want to do this in the future. My main point is
that we still need to settle on the api. And from experience, not make more
work for ourselves. Also, I don't see a need to enforce the class as
static, an instance would still be able to call those methods anyway. No?